### ISSUE TITLE
build: don't blanket ban parallel tests on kdbtests

### ISSUE THREAD

**Issue by ingwinlu** [ElektraInitiative/libelektra#1935]
Currently we flag every kdbtest as `NO_SERIAL`. However this might not be necessary as only those tests that could influence others should be flagged.

Providing the means via a test_macro to quickly add the flag if needed would allow to speed up tests even more.

**Comment by markus2330** [comment]
Thank you for writing this up! I was also considering writing such an issue.

Its highly likely that we have "kdbtests" that actually do not use KDB and thus should have neither the flag "kdbtests" nor "NO_SERIAL".

If a test needs "kdbtests", however, it should have "NO_SERIAL", too. (Otherwise tests might fail due to conflicts in concurrent access.)

If there is time, one could also reevaluate if the memleak flag is always correct.

Basically it is trivial to find out if "kdbtests" and "memleak" are tagged correctly.

1. to find out about "kdbtests" simply disallow your user write access to all namespaces. If a "kdbtests" still passes, it was tagged wrongly.
2. to find out about "memleak" simply run the test with ctest `-T memcheck`. If it passes, it was wrongly tagged.
